### DISCUSSION TITLE
Dialog and Dropdown Menu not working properly

### DISCUSSION THREAD

**Discussion by Wooks08** [shadcn-ui/ui#3942] [Q&A]
I wanted to add a Theme Toggle for my site. It worked at the beginning but then stopped. I didn't care much about it because I thought I'd fix it later. But when lately I tried to add Popup or Dialog it does not show as well, so it seemed like this is a problem related to all components that render on top of the site. But I tried adding buttons and it showed that it's fault of buttons which are used in each of them. Any suggestions?

*And yes I do have it set as a use client*

**Comment by mewtwo0719** [comment]
Can you provide more information about your project configuration and possibly console errors?

**Reply by hal-enlighten** [reply]
im getting this when using dropdown 

Cannot destructure property 'name' of '_ref' as it is null.

**Reply by mtgred** [reply]
Had the same "Cannot destructure property 'name' of '_ref' as it is null" when using the shadcn dropdown menu.

Upgrading React from 18.2 to 18.3.1 solved the issue.

Alternatively, using the `dropdown-menu.tsx` generated in august 2024 from another project also works with React 18.2.
